testgen/src/header.rs

```diff
@@ -64,6 +68,29 @@ where
     m_secs.serialize(serializer)
 }
 
+// Serialize and deserialize the `Option<AppHash>`, delegating to the `AppHash`
+// serialization/deserialization into/from hexstring.
+mod app_hash_serde {
+    use super::*;
+    pub fn deserialize<'de, D>(deserializer: D) -> Result<Option<AppHash>, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        tendermint::serializers::apphash::deserialize(deserializer).map(Some)
+    }
+
+    pub fn serialize<S>(value: &Option<AppHash>, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        if value.is_none() {
+            serializer.serialize_none()
+        } else {
+            tendermint::serializers::apphash::serialize(value.as_ref().unwrap(), serializer)
+        }
```

Comment:
You no longer need to handle the `None` case since it is skipped by way of the `skip_serializing_if` parameter, so might as well do:

```suggestion
        tendermint::serializers::apphash::serialize(value.as_ref().unwrap(), serializer)
```